Orange/tests/test_tab_reader.py

```diff
@@ -226,3 +225,15 @@ def test_no_metadata(self):
         TabReader.write_table_metadata(fname, table)
         self.assertFalse(path.isfile(fname + ".metadata"))
         shutil.rmtree(tempdir)
+
+    def test_had_metadata_now_there_is_none(self):
+        tempdir = tempfile.mkdtemp()
+        table = Table("titanic")
+        table.attributes["a"] = "aa"
+        fname = path.join(tempdir, "out.tab")
+        TabReader.write_table_metadata(fname, table)
+        self.assertTrue(path.isfile(fname + ".metadata"))
+        del table.attributes["a"]
+        TabReader.write_table_metadata(fname, table)
+        self.assertFalse(path.isfile(fname + ".metadata"))
+        shutil.rmtree(tempdir)
```

Comment:
I guess you should do this in `finally` (of a `try` put under `tempdir = ...`) to ensure that the directory is removed even when the test fails?